pkg/compactor/bucket_compactor.go

```diff
@@ -361,7 +362,13 @@ func (c *BucketCompactor) runCompactionJob(ctx context.Context, job *Job) (shoul
 	uploadBegin := time.Now()
 	uploadedBlocks := atomic.NewInt64(0)
 
+	if err = verifyCompactedBlocksTimeRanges(ctx, compIDs, toCompactMinTime.UnixMilli(), toCompactMaxTime.UnixMilli(), c.blockSyncConcurrency, subDir); err != nil {
+		level.Error(jobLogger).Log("msg", err)
```

Comment:
This can log message like `msg="failed to read meta.json from ...: file not found"`, which will very little context.

It would also be nice to distringuish between errors during verification (eg. corrupted meta.json file), and verification failure (wrong dates in the block). We could do that by returning not just error, but `string, error`, where string would be verification failure or empty. Alternatively we could handle logging and metric increment in the method.